docs/source/usage/workflows/memoryPerDevice.py

```diff
@@ -26,90 +27,113 @@
+    if cells_to_spread >= Ny:
+        GPU_rows[ii] = Ny
+        cells_to_spread -= Ny
+    else:
+        GPU_rows[ii] = cells_to_spread
+        break
+# remove vacuum cells from the front rows
+extra_cells = vacuum_cells
+for ii, _ in enumerate(GPU_rows):
+    if extra_cells >= Ny:
+        GPU_rows[ii] = 0
+        extra_cells -= Ny
+    else:
+        GPU_rows[ii] -= extra_cells
+        break
 
 pmc = MemoryCalculator(Nx, Ny, Nz)
 
-target_x = Nx  # full transversal dimension of the device
-target_y = target_cells  # only the first row of devices holds the target
-target_z = Nz
-
 # typical number of particles per cell which is multiplied later for
 # each species and its relative number of particles
 N_PPC = 6
-
 # conversion factor to megabyte
 megabyte = 1.0 / (1024 * 1024)
 
-print("Memory requirement per device:")
-# field memory per device
-field_device = pmc.mem_req_by_fields(Nx, Ny, Nz, field_tmp_slots=2,
-                                     particle_shape_order=2)
-print("+ fields: {:.2f} MB".format(
-      field_device * megabyte))
-
-# electron macroparticles per supercell
-e_PPC = N_PPC * (
-    # H,C,N pre-ionization - higher weighting electrons
-    3 \
-    # electrons created from C ionization
-    + (6 - 2) \
-    # electrons created from N ionization
-    + (7 - 2)
-)
-# particle memory per device - only the target area contributes here
-e_device = pmc.mem_req_by_particles(
-    target_x, target_y, target_z,
-    num_additional_attributes=0,
-    particles_per_cell=e_PPC
-)
-H_device = pmc.mem_req_by_particles(
-    target_x, target_y, target_z,
-    # no bound electrons since H is preionized
-    num_additional_attributes=0,
-    particles_per_cell=N_PPC
-)
-C_device = pmc.mem_req_by_particles(
-    target_x, target_y, target_z,
-    num_additional_attributes=1,
-    particles_per_cell=N_PPC
-)
-N_device = pmc.mem_req_by_particles(
-    target_x, target_y, target_z,
-    num_additional_attributes=1,
-    particles_per_cell=N_PPC
-)
-# memory for calorimeters
-cal_device = pmc.mem_req_by_calorimeter(
-    n_energy=1024, n_yaw=360, n_pitch=1
-) * 2  # electrons and protons
-# memory for random number generator states
-rng_device = pmc.mem_req_by_rng(Nx, Ny, Nz)
-
-print("+ species:")
-print("- e: {:.2f} MB".format(e_device * megabyte))
-print("- H: {:.2f} MB".format(H_device * megabyte))
-print("- C: {:.2f} MB".format(C_device * megabyte))
-print("- N: {:.2f} MB".format(N_device * megabyte))
-print("+ RNG states: {:.2f} MB".format(
-    rng_device * megabyte))
-print(
-    "+ particle calorimeters: {:.2f} MB".format(
-        cal_device * megabyte))
-
-mem_sum = (field_device + e_device + H_device + C_device +
-           N_device + rng_device + cal_device)
-print("Required memory per device: {:.2f} MB".format(
-    mem_sum * megabyte))
+target_x = Nx  # full transverse dimension of the GPU
+target_z = Nz
+
+sx = lambda n: {1: "st", 2: "nd", 3: "rd"}.get(n if n < 20 else int(str(n)[-1]), "th")
```

Comment:
The Travis CI suite fails because it doesn't meet the PEP8 standard to assign a `lambda` function to a memory object.
Maybe transform it into a function definition with `def`. ^^

It's only valid until 20 now but that's okay for the example.

Reply:
>  It's only valid until 20 now but that's okay for the example.

I'm pretty sure that it works fine until 111, that should be enough. IAnyway, it's not the most crucial part  :sweat_smile:

Reply:
Oh, ha, I chose to overlook the `else` part. You're right. :P